src/main/scala/cognite/spark/v1/StreamIterator.scala

```diff
@@ -0,0 +1,105 @@
+package cognite.spark.v1
+
+import java.util.concurrent.{ArrayBlockingQueue, ConcurrentHashMap, Executors}
+
+import cats.effect.{Concurrent, IO}
+import fs2.{Chunk, Stream}
+import org.apache.spark.sql.Row
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object StreamIterator {
+  type EitherQueue = ArrayBlockingQueue[Either[Throwable, Chunk[Row]]]
+
+  def apply(stream: Stream[IO, Row], queueBufferSize: Int, toId: Option[Row => String] = None)(
```

Comment:
`queueBufferSize` = `numBatchesInQueue` so it should be closer to the number of batches running at any time.

Reply:
@Tapped not sure if I understood what you meant by this, maybe I'm doing it already, otherwise could you suggest what we should change it to?